postgresql/resource_postgresql_grant.go

```diff
@@ -693,6 +693,13 @@ func revokeRolePrivileges(txn *sql.Tx, d *schema.ResourceData) error {
 		// Query is empty, don't run anything
 		return nil
 	}
+
+	// Obtain a lock to avoid `Error: could not execute revoke query: pq: tuple concurrently updated`
+	schema := d.Get("schema").(string)
+	if err := pgLockSchema(txn, schema); err != nil {
+		return err
+	}
```

Comment:
Should we do it next to where we call `pgLockRole` (e.g. around line 171) ? 

And actually  as `schema` is optional if `postgresql_grant` is used at database level, we should put this in the else of  `if objectType == "database" { ` line 176

```go
	if objectType == "database" {
		if err := pgLockDatabase(txn, database); err != nil {
			return err
		}
	} else {
		// Obtain a lock to avoid `Error: could not execute revoke query: pq: tuple concurrently updated`
		schema := d.Get("schema").(string)
		if err := pgLockSchema(txn, schema); err != nil {
			return err
		}
	}
```